digitalocean/partnerinterconnectattachment/resource_partner_interconnect_attachment_test.go

```diff
@@ -0,0 +1,165 @@
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_partner_interconnect_attachment.foobar", "created_at"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_partner_interconnect_attachment.foobar", "status"),
+				),
+			},
+			{
+				Config: partnerInterconnectAttachmentUpdateConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanPartnerInterconnectAttachmentExists("digitalocean_partner_interconnect_attachment.foobar", &partnerInterconnectAttachment),
+					resource.TestCheckResourceAttr(
+						"digitalocean_partner_interconnect_attachment.foobar", "name", updatePartnerInterconnectAttachmentName),
+				),
+			},
+			{
+				Config: partnerInterconnectAttachmentVPCUpdateConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanPartnerInterconnectAttachmentExists("digitalocean_partner_interconnect_attachment.foobar", &partnerInterconnectAttachment),
+					resource.TestCheckResourceAttr(
+						"digitalocean_partner_interconnect_attachment.foobar", "vpc_ids.#", "2"),
+					resource.TestCheckResourceAttrPair(
+						"digitalocean_partner_interconnect_attachment.foobar", "vpc_ids.0", "digitalocean_vpc.vpc1", "id"),
+					resource.TestCheckResourceAttrPair(
+						"digitalocean_partner_interconnect_attachment.foobar", "vpc_ids.1", "digitalocean_vpc.vpc3", "id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckDigitalOceanPartnerInterconnectAttachmentDestroy(s *terraform.State) error {
+	client := acceptance.TestAccProvider.Meta().(*config.CombinedConfig).GodoClient()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "digitalocean_partner_interconnect_attachment" {
+			continue
+		}
+
+		_, _, err := client.PartnerInterconnectAttachments.Get(context.Background(), rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Partner Interconnect Attachment still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckDigitalOceanPartnerInterconnectAttachmentExists(n string, partnerInterconnectAttachment *godo.PartnerInterconnectAttachment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Partner Interconnect Attachment not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("Partner Interconnect Attachment ID is not set")
+		}
+
+		client := acceptance.TestAccProvider.Meta().(*config.CombinedConfig).GodoClient()
+		foundPartnerInterconnectAttachment, _, err := client.PartnerInterconnectAttachments.Get(context.Background(), rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("error fetching Partner Interconnect Attachment (%s): %s", rs.Primary.ID, err)
+		}
+
+		*partnerInterconnectAttachment = *foundPartnerInterconnectAttachment
+
+		return nil
+	}
+}
+
+const testAccCheckDigitalOceanPartnerInterconnectAttachmentConfig_Basic = `
+resource "digitalocean_vpc" "vpc1" {
+  name   = "vpc1"
```

Comment:
The VPC names used in these test configs should also be randomly generated to prevent clashes with any existing VPCs on the account used for testing.